platforms/core-configuration/model-core/src/main/java/org/gradle/api/internal/provider/AbstractCollectionProperty.java

```diff
@@ -354,11 +360,6 @@ public NoValueSupplier(Value<? extends C> value) {
             this.value = value.asType();
         }
 
-        @Override
-        public CollectionSupplier<T, C> absentIgnoring() {
-            return Cast.uncheckedCast(emptySupplier());
```

Comment:
:x: without this `append` to a cleared property doesn't work. A test that reproduces the issue:
```groovy
        property.set(null as Iterable)
        property.append("foo")

        expect:
        assertValueIs(["foo"])
```